core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/toDataFrame.kt

```diff
@@ -251,11 +251,9 @@ internal fun convertToDataFrame(
         val kClass = returnType.classifier as KClass<*>
         val fieldKind = returnType.getFieldKind()
 
-        val shouldCreateValueCol = (
-            maxDepth <= 0 &&
-                !fieldKind.shouldBeConvertedToFrameColumn &&
-                !fieldKind.shouldBeConvertedToColumnGroup
-        ) ||
+        val keepSubtree =
+            maxDepth <= 0 && !fieldKind.shouldBeConvertedToFrameColumn && !fieldKind.shouldBeConvertedToColumnGroup
```

Comment:
Is it fine that in this logic exp `maxDepth <= 0` and in impl/api/toDataFrame is `depth>maxDepth`?

Reply:
I think so. I didn't want to follow the same logic in compiler implementation because incrementing depth in recursive calls seems decreasing maxDepth. Maybe i should re-evaluate and change one to work as the other